### PR TITLE
Change registry domain & mirrors based on region

### PR DESCRIPTION
**Pull request by wtty-fool**
Towards https://github.com/giantswarm/giantswarm/issues/11571
Initial discussion https://github.com/giantswarm/giantswarm/issues/12100
Closes https://github.com/giantswarm/giantswarm/issues/12149

## Checklist
- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: helm/aws-operator/templates/configmap.yaml [hunk 1]
```diff
@@ -45,15 +45,25 @@ data:
           clusterDomain: '{{ .Values.Installation.V1.Guest.Kubernetes.ClusterDomain }}'
           networkSetup:
             docker:
+              {{- if hasPrefix .Values.Installation.V1.Provider.AWS.Region 'cn-' }}
```
Comment (wtty-fool) [inline helm/aws-operator/templates/configmap.yaml:48]: `hasPrefix` documentation: https://helm.sh/docs/chart_template_guide/function_list/#hasprefix-and-hassuffix

**Comment by wtty-fool** [comment]
Tested on _gauss_, TC created, nodes come up with `quay.io` and either of `docker.io`/`azurecr.io` disabled.

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 
 - Add `--service.registry.mirrors` flag for setting registry mirror domains.
+- Replace `--service.registrydomain` with `--service.registry.domain`.
```
Comment (xh3b4sd) [inline CHANGELOG.md:14]: This looks like it should not be there.

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 
 - Add `--service.registry.mirrors` flag for setting registry mirror domains.
+- Make registry domain & mirrors configurable based on region
```
Comment (xh3b4sd) [inline CHANGELOG.md:14]:
```suggestion
- Make registry domain & mirrors configurable based on region.
```

**Comment by wtty-fool** [comment]
@giantswarm/team-biscuit Kind reminder to take a look here :smile:

File: service/internal/cloudconfig/tcnp.go [hunk 4]
```diff
@@ -162,6 +162,7 @@ func (t *TCNP) NewTemplates(ctx context.Context, obj interface{}) ([]string, err
 			registryDomain: t.config.RegistryDomain,
 		}
 		params.Kubernetes.Kubelet.CommandExtraArgs = kubeletExtraArgs
+		params.RegistryMirrors = t.config.RegistryMirrors
```
Comment (kopiczko) [inline service/internal/cloudconfig/tcnp.go:165]: Merging master will remove this change from this PR.

**Pull request merged**